plugin/hosts/hostsfile.go

```diff
@@ -140,18 +140,29 @@ func (h *Hostsfile) Parse(file io.Reader) {
 			}
 			switch ver {
 			case 4:
-				hsv4[name] = append(hsv4[name], addr)
+				hmap.byNameV4[name] = append(hmap.byNameV4[name], addr)
 			case 6:
-				hsv6[name] = append(hsv6[name], addr)
+				hmap.byNameV6[name] = append(hmap.byNameV6[name], addr)
 			default:
 				continue
 			}
-			is[addr.String()] = append(is[addr.String()], name)
+			hmap.byAddr[addr.String()] = append(hmap.byAddr[addr.String()], name)
 		}
 	}
-	h.byNameV4 = hsv4
-	h.byNameV6 = hsv6
-	h.byAddr = is
+
+	if override != nil {
```

Comment:
I would just return here if `override == nil` and then do the override thing below